### PR TITLE
Add Hungarian localization

### PR DESCRIPTION
**Pull request by tevelee**

### CODE REVIEW

File: FormatterKit/TTTTimeIntervalFormatter.m [hunk 1]
```diff
@@ -509,6 +512,35 @@ - (NSString *)itRelativeDateStringForComponents:(NSDateComponents *)components {
...
+        [stringComponents addObject:[NSString stringWithFormat:@"%d napja", -1 * (int)[components day]]];
+    } else if ([components hour] < 0) {
+        [stringComponents addObject:[NSString stringWithFormat:@"%d órája", -1 * (int)[components hour]]];
+    } else if ([components minute] < 0) {
+        [stringComponents addObject:[NSString stringWithFormat:@"%d perce", -1 * (int)[components minute]]];
+    } else if ([components second] < 0) {
+        [stringComponents addObject:[NSString stringWithFormat:@"%d másodperce", -1 * (int)[components second]]];
+    }
+    
+    if ([stringComponents count] > 0) {
```
Comment (yas375) [inline FormatterKit/TTTTimeIntervalFormatter.m]: It will always have 0 or 1 elements because you are adding elements to the array in `else if`...

File: FormatterKit/TTTTimeIntervalFormatter.m [hunk 2]
```diff
@@ -509,6 +512,35 @@ - (NSString *)itRelativeDateStringForComponents:(NSDateComponents *)components {
     return nil;
 }
 
+- (NSString *)huRelativeDateStringForComponents:(NSDateComponents *)components {
```
Comment (yas375) [inline FormatterKit/TTTTimeIntervalFormatter.m]: Are you sure this method is needed for Hungarian localization? The code you have doesn't seem to be related to localized idiomatic deictic expressions...

File: FormatterKit/TTTOrdinalNumberFormatter.m [hunk 3]
```diff
@@ -112,10 +112,6 @@ - (NSString *)caOrdinalIndicatorStringFromNumber:(NSNumber *)number {
     }
 }
 
-- (NSString *)deOrdinalIndicatorStringFromNumber:(__unused NSNumber *)number {
```
Comment (yas375) [inline FormatterKit/TTTOrdinalNumberFormatter.m]: Good catch! I've cherry-picked this commit to `master`: 6e9163d6055c5a3de63e512dd4a79c64acd42736

File: FormatterKit/TTTTimeIntervalFormatter.m (hunk 1 above)
Comment (tevelee) [inline FormatterKit/TTTTimeIntervalFormatter.m]: Thanks, good catch! :+1:

File: FormatterKit/TTTTimeIntervalFormatter.m (hunk 2 above)
Comment (tevelee) [inline FormatterKit/TTTTimeIntervalFormatter.m]: You are right, thanks for the observation. I corrected this behaviour in my next commit.

File: FormatterKit/TTTTimeIntervalFormatter.m [hunk 4]
```diff
@@ -21,6 +21,7 @@
 // THE SOFTWARE.
 
 #import "TTTTimeIntervalFormatter.h"
+#import "TTTArrayFormatter.h"
```
Comment (yas375) [inline FormatterKit/TTTTimeIntervalFormatter.m]: This is probably is not needed anymore.

File: FormatterKit/TTTTimeIntervalFormatter.m [hunk 5]
```diff
@@ -184,7 +185,12 @@ - (NSString *)stringForTimeIntervalFromDate:(NSDate *)startingDate
     if (string) {
         if (seconds > 0) {
             if ([self.pastDeicticExpression length]) {
-                string = [NSString stringWithFormat:self.deicticExpressionFormat, string, self.pastDeicticExpression];
+                NSString *languageCode = [self.locale objectForKey:NSLocaleLanguageCode];
+                if ([languageCode isEqualToString:@"hu"]) {
+                    string = [self huPastDeicticExpression:components];
```
Comment (yas375) [inline FormatterKit/TTTTimeIntervalFormatter.m]: It looks like this makes the formatter completely ignore its `deicticExpressionFormat` property for Hungarian. I'm not sure this is the right way to achieve what you are trying to achieve...

Comment (tevelee) [inline FormatterKit/TTTTimeIntervalFormatter.m]: I had the same feeling, this language looks like a big distinction and has completely new concepts. Another solution would be to introduce the "past" version of calendar units, which would useless on other languages. 
Do you have any alternatives in mind? It's true that it cannot be solved using a static `deicticExpressionFormat`...

File: FormatterKit/TTTTimeIntervalFormatter.m (hunk 4 above)
Comment (tevelee) [inline FormatterKit/TTTTimeIntervalFormatter.m]: Thanks

**Comment by yas375** [comment]
I've just recently started to maintain this library. I'm a little worry about making changes like this without unit tests. At the same time I've just recently started to add tests to the library and currently it's not ready to execute tests using different locales. This is on my todo list.

As you already noticed, the example app doesn't actually support localizations so it's even harder to test with the example app. I'm planning to restructure the project so
1. it support CocoaPods and Carhage
2. it makes possible to run tests specific to some locales
3. it's possible to edit localization files from Xcode's project
4. example app supports different localizations

I really appreciate your contribution and here is what I'd like to propose to do.

1. I'm happy to merge `Localizations/hu.lproj/FormatterKit.strings` now
2. after that we can create an issue where you are more than welcome to provide examples specific for Hungarian localization.
3. once I'm ready with project structure update, we can convert your examples into unit tests. Hopefully adding unit tests for some other locales as well. I.e. there are some known issues with Russian localizations (#169).
4. then we can work on a fix specific to Hungarian language with a confidence that likely we are not breaking anything else.

Does this make sense?

**Comment by tevelee** [comment]
Makes perfect sense, I really like this approach! I noticed the same in the example app, it's not comprehensive at the moment. Project structure update + unit tests would add an extra confidence level and it would be great to insert language-specific fixes easily.
I'd be happy to help with this process. I'll create the issue as a first step, and provide some examples. I also have some suggestions not just regarding time-formatting.
Thanks for all this @yas375!

**Comment by yas375** [comment]
@tevelee I've cherry-picked your `Localizations/hu.lproj/FormatterKit.strings`. e18af1c

Thanks again!
